### PR TITLE
Alb stop

### PR DESCRIPTION
**Pull request by kristianiliev1**
## Description
Add option to remove the application gateway in order we need to use external one.

## Related Issues
[GDB-10886](https://ontotext.atlassian.net/browse/GDB-10886)

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

[GDB-10886]: https://ontotext.atlassian.net/browse/GDB-10886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: certificate2.pfx
Comment (viktor-ribchev) [inline certificate2.pfx:1]: No no no

File: main.tf [hunk 1]
```diff
@@ -222,10 +225,11 @@ module "monitoring" {
   location             = var.location
   node_count           = var.node_count
 
-  web_test_availability_request_url = module.application_gateway.public_ip_address_fqdn
-  web_test_geo_locations            = var.web_test_geo_locations
-  web_test_ssl_check_enabled        = var.web_test_ssl_check_enabled
-  graphdb_external_address_fqdn     = var.graphdb_external_address_fqdn != null ? var.graphdb_external_address_fqdn : module.application_gateway.public_ip_address_fqdn
+  web_test_availability_request_url = var.disable_agw ? null : module.application_gateway[0].public_ip_address_fqdn
+  #web_test_availability_request_url = module.application_gateway.public_ip_address_fqdn
```
Comment (viktor-ribchev) [inline main.tf:229]: Remove the commented variable

File: main.tf [hunk 2]
```diff
@@ -270,14 +274,15 @@ module "graphdb" {
   graphdb_outbound_address_prefixes    = var.outbound_allowed_address_prefixes
 
   # Gateway
-  application_gateway_backend_address_pool_ids = [module.application_gateway.gateway_backend_address_pool_id]
+  #application_gateway_backend_address_pool_ids = [module.application_gateway.gateway_backend_address_pool_id]
```
Comment (viktor-ribchev) [inline main.tf:277]: Remove the commented variable

File: modules/gateway/variables.tf [hunk 3]
```diff
@@ -55,6 +55,12 @@ variable "gateway_enable_private_access" {
   type        = bool
 }
 
+variable "disable_agw" {
+  description = "Disables the application gateway"
```
Comment (viktor-ribchev) [inline modules/gateway/variables.tf:59]: Which Application Gateway?

File: modules/graphdb/main.tf [hunk 4]
```diff
@@ -91,11 +91,14 @@ resource "azurerm_linux_virtual_machine_scale_set" "graphdb" {
...
-      subnet_id                                    = var.graphdb_subnet_id
-      application_gateway_backend_address_pool_ids = var.application_gateway_backend_address_pool_ids
-      application_security_group_ids               = [azurerm_application_security_group.graphdb_vmss.id]
+      name      = "${var.resource_name_prefix}-ip-config"
+      primary   = true
+      subnet_id = var.graphdb_subnet_id
+
+      application_gateway_backend_address_pool_ids = var.disable_agw ? [] : compact(var.application_gateway_backend_address_pool_ids)
+
+      # Ensure no null values by using coalesce or compact
```
Comment (viktor-ribchev) [inline modules/graphdb/main.tf:100]: ChatGPT?

File: README.md [hunk 5]
```diff
@@ -135,14 +135,15 @@ az vm image terms accept --offer graphdb-ee --plan graphdb-byol --publisher onto
...
 | gateway\_private\_link\_service\_network\_policies\_enabled | Enable or disable private link service network policies | `string` | `false` | no |
 | gateway\_backend\_port | Backend port for the Application Gateway rules | `number` | `7201` | no |
 | gateway\_probe\_interval | Interval in seconds between the health probe checks | `number` | `10` | no |
 | gateway\_probe\_timeout | Timeout in seconds for the health probe checks | `number` | `1` | no |
 | gateway\_probe\_threshold | Number of consecutive health checks to consider the probe passing or failing | `number` | `2` | no |
-| tls\_certificate\_path | Path to a TLS certificate that will be imported in Azure Key Vault and used in the Application Gateway TLS listener for GraphDB. | `string` | `null` | no |
-| tls\_certificate\_password | TLS certificate password for password protected certificates. | `string` | `null` | no |
+| tls\_certificate\_path | Path to a TLS certificate that will be imported in Azure Key Vault and used in the Application Gateway TLS listener for GraphDB. | `string` | n/a | yes |
```
Comment (viktor-ribchev) [inline README.md:145]: Leave a comment somewhere in the README that either  `tls_certificate_path` or `tls_certificate_id` is required

File: main.tf [hunk 6]
```diff
@@ -158,6 +158,7 @@ module "tls" {
 # Creates a public IP address and an Application Gateway for forwarding internet traffic to the GraphDB proxies/instances
 module "application_gateway" {
   source = "./modules/gateway"
+  count  = var.disable_agw ? 0 : 1
```
Comment (viktor-ribchev) [inline main.tf:161]: All other modules use variables such as `deploy_something` why disable?

File: main.tf [hunk 7]
```diff
@@ -158,6 +158,7 @@ module "tls" {
 # Creates a public IP address and an Application Gateway for forwarding internet traffic to the GraphDB proxies/instances
```
Comment (viktor-ribchev) [inline main.tf:158]: Have you tested deploy with API Gateway then switch to another one and vice versa ?

File: modules/gateway/outputs.tf [hunk 8]
```diff
@@ -24,10 +24,11 @@ output "public_ip_address_fqdn" {
...
+  value       = var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 ? azurerm_application_gateway.graphdb-private[0].id : !var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-public) > 0 ? azurerm_application_gateway.graphdb-public[0].id : null
 }
 
 output "gateway_backend_address_pool_id" {
   description = "Identifier of the application gateway backend address pool"
-  value       = var.gateway_enable_private_access ? one(azurerm_application_gateway.graphdb-private[0].backend_address_pool).id : one(azurerm_application_gateway.graphdb-public[0].backend_address_pool).id
+  value       = var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 ? length(azurerm_application_gateway.graphdb-private[0].backend_address_pool) > 0 ? one(azurerm_application_gateway.graphdb-private[0].backend_address_pool).id : null : !var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-public) > 0 ? length(azurerm_application_gateway.graphdb-public[0].backend_address_pool) > 0 ? one(azurerm_application_gateway.graphdb-public[0].backend_address_pool).id : null : null
```
Comment (viktor-ribchev) [inline modules/gateway/outputs.tf:32]: Can you simplify this?

**Comment by viktor-ribchev** [comment]
https://namingconvention.org/git/pull-request-naming

File: modules/gateway/variables.tf (hunk 3 above)
Comment (kristianiliev1) [reply modules/gateway/variables.tf:59]: Changed for clarity

File: README.md (hunk 5 above)
Comment (kristianiliev1) [reply README.md:145]: There was a separate pull request for that change - https://github.com/Ontotext-AD/terraform-azure-graphdb/pull/87

**Pull request closed without merging**